system/metrics/sinks/prometheus/prometheusSink.cpp

```diff
@@ -104,7 +104,10 @@ void PrometheusMetricSink::toPrometheusMetrics(const std::vector<std::shared_ptr
 
     for (auto &pMetric: reportMetrics)
     {
-        const std::string & name = pMetric->queryName();
+        std::string name = pMetric->queryName();
+
+        '.' is a known char used in HPCC metric names but invalid in Prometheus
```

Comment:
I think you meant this to be a comment!
Also, I have a memory that we expected '.' to be replaced by '_' in the prometheus output.  Would that make sense?

Reply:
Absolutely, fixing right away.
We can replace with underscores, chose colon because underscores seemed to be used within the various segments of the metric name, thought we might have wanted a separate delimiter between the segments. @ghalliday